### ISSUE TITLE
Fix S2325 FP: "Make 'MethodName' a static method." should not apply to overriding instance methods

### ISSUE THREAD

**Issue by richardissimo** [SonarSource/sonar-dotnet#9651]
### Description

S2325 Make 'MethodName' a static method.
If I have an abstract class with a virtual method, and the derived class has an implementation which _could_ be made static, this should not trigger this rule, because it is overriding the virtual.

### Repro steps

``` csharp
    public abstract class DemoProblem
    {
        protected virtual bool IsAllowed()
        {
            return true;
        }
    }

    public class DerivedClass : DemoProblem
    {
        public bool IsAllowed() // Triggers S2325
        {
            return false;
        }
    }
```

### Expected behavior

Should not trigger S2325

### Actual behavior

triggers S2325

### Known workarounds

Suppress rule

### Related information

* SonarAnalyzer.CSharp 9.32.0.97167
* Visual Studio version 17.11.0
* MSBuild / dotnet version <TargetFrameworks>netstandard2.0;net452;net48;net6.0;net8.0</TargetFrameworks>
* Operating System Windows 10

**Comment by richardissimo** [comment]
Hmm. My made up example isn't exactly what I had for real, and I can see in that example, the DerivedClass implementation should override. I'll see if I can find a better example, closer to my problem.

**Comment by richardissimo** [comment]
No, I'm withdrawing this. This is more like what the actual code looks like, so fair enough, that can be made static.
```
    public abstract class Rule
    {
        protected string SimplifyWhitespaceExpression(string expression)
        {
            // Implementation isn't relevant, but it can be made static
        }
```
